mlflow/transformers.py

```diff
@@ -1601,6 +1596,54 @@ def _coerce_exploded_dict_to_single_dict(self, data):
         else:
             return data
 
+    @staticmethod
+    def _strip_input_from_response_in_instruction_pipelines(input_data, output, output_key):
+        """
+        Parse the output from instruction pipelines to conform with other text generator
+        pipeline types and remove line feed characters and other confusing outputs
+        """
+        replacements = {"\n\n": " ", "A:": ""}
+
+        def extract_response_data(data_out):
+            if all(isinstance(x, dict) for x in data_out):
+                return [elem[output_key] for elem in data_out][0]
+            elif all(isinstance(x, list) for x in data_out):
+                return [elem[output_key] for coll in data_out for elem in coll]
+            else:
+                raise MlflowException(
+                    "Unable to parse the pipeline output. Expected List[Dict[str,str]] or "
+                    f"List[List[Dict[str,str]]] but got {type(data_out)} instead."
+                )
+
+        output = extract_response_data(output)
+
+        def trim_input(data_in, data_out):
+            # NB: the '\n\n' pattern is exclusive to specific InstructionalTextGenerationPipeline
+            # types that have been loaded as a plain TextGenerator. The structure of these
+            # pipelines will precisely repeat the input question immediately followed by 2 carriage
+            # return statements, followed by the start of the response to the prompt. We only
+            # want to left-trim these types of pipelines output values.
+            if data_out.startswith(data_in + "\n\n"):
+                output_string = data_out[len(data_in) :].strip()
```

Comment:
Should we keep the input question if a user requests Dolly to do so?

https://github.com/databrickslabs/dolly/blob/0eadcb7b0648d496d67243a7d572b413560be661/training/generate.py#L207

```
            # If the full text is requested, then append the decoded text to the original instruction.
            # This technically isn't the full text, as we format the instruction in the prompt the model has been
            # trained on, but to the client it will appear to be the full text.
            if return_full_text:
                decoded = f"{instruction_text}\n{decoded}"
```

Reply:
Added an inference_config pop functionality to check for disabling the prompt inclusion.